tests/test_gates_channels.py

```diff
@@ -209,7 +209,7 @@ def test_pauli_noise_channel(backend, pauli_order):
     )
     gate = gates.X(1)
     target_state = backend.apply_gate_density_matrix(gate, np.copy(initial_state), 2)
-    target_state = 0.3 * target_state + 0.7 * initial_state
+    target_state = 0.3 * target_state + 0.7 * backend.cast(initial_state)
```

Comment:
this cast should not be necessary since `initial_state` is supposedly generated in the correct backend